starter-core/src/main/java/io/micronaut/starter/feature/micrometer/MicrometerAnnotations.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2017-2022 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.starter.feature.micrometer;
+
+import io.micronaut.starter.application.ApplicationType;
+import io.micronaut.starter.application.generator.GeneratorContext;
+import io.micronaut.starter.build.dependencies.Dependency;
+import io.micronaut.starter.feature.Category;
+import io.micronaut.starter.feature.Feature;
+import io.micronaut.starter.feature.FeatureContext;
+import io.micronaut.starter.feature.other.Management;
+import io.micronaut.starter.feature.server.MicronautServerDependent;
+import io.micronaut.starter.util.NameUtils;
+import jakarta.inject.Singleton;
+
+@Singleton
+public class MicrometerAnnotations implements Feature, MicronautServerDependent {
+
+    private final Core core;
+    private final Management management;
+
+    public MicrometerAnnotations(Core core, Management management) {
+        this.core = core;
+        this.management = management;
+    }
+
+    @Override
+    public String getName() {
+        return "micrometer-annotation";
+    }
+
+    @Override
+    public String getTitle() {
+        return NameUtils.getNaturalName(io.micronaut.core.naming.NameUtils.dehyphenate(getName()));
+    }
+
+    @Override
+    public String getDescription() {
+        return "Adds support for Micrometer annotations (@Timed and @Counted)";
+    }
+
+    @Override
+    public void processSelectedFeatures(FeatureContext featureContext) {
+        if (!featureContext.isPresent(Core.class)) {
+            featureContext.addFeature(core);
+        }
+        if (!featureContext.isPresent(Management.class)) {
+            featureContext.addFeature(management);
+        }
+    }
+
+    @Override
+    public void apply(GeneratorContext generatorContext) {
+        generatorContext.addDependency(Dependency.builder()
+                .groupId("io.micronaut.micrometer")
+                .artifactId("micronaut-micrometer-annotation")
+                .versionProperty("micronaut.micrometer.version")
```

Comment:
is the property necessary?

Reply:
Yes, it is needed for Maven builds.

Reply:
Then, don't you need to do it only if it's a Maven build?

Reply:
The full version of that section is:
```
generatorContext.addDependency(Dependency.builder()
                .groupId("io.micronaut.micrometer")
                .artifactId("micronaut-micrometer-annotation")
                .versionProperty("micronaut.micrometer.version")
                .annotationProcessor());
```

Without the `versionProperty` the Maven Rocker template fails to generate the pom correctly, because for annotations a version is required.

Reply:
No, what I mean is that, don't you need to wrap that block in:

```java
if (!generatorContext.getBuildTool().isGradle()) {
  ...
}
```

Reply:
Alvaro, missed the point of your question. The version property seems to be only added for Maven and similar to other code in Starter.